### ISSUE TITLE
Current build shows wrong search results.

### ISSUE THREAD

**Issue by junedev** [christian-fei/Simple-Jekyll-Search#51]
I was using the version of this search that shipped with the theme I am using for my blog. https://github.com/jacobtomlinson/carte-noire/tree/gh-pages/js 
Everything was fine. 

Now I saw that there is an uncompressed build and I wanted to use that one to make minor changes. When plugging in (changing dataSource to json, removing .init) the search shows nearly all blog posts as search results even though the search word is not contained. Same behaviour for the current minified version I pulled with bower.

Example: http://junedev.github.io/search/
Searching for "angular" returns a lot of blog posts. None of them has "angular" in the text.
Code: https://github.com/junedev/junedev.github.io/blob/master/search.html

Any hints what changes might cause this to break compared the old version?

**Comment by junedev** [comment]
I found that the old version that worked for me was not actually minified very much. I could make my edits there and I am fine now. Feel free to close the issue if no one else reports problems with the current build.

**Comment by christian-fei** [comment]
Aaah, alright, problably it was a faulty version. Tried now for an hour to reproduce the issue, it was indeed very strange :)
Thanks for using the library!

**Comment by junedev** [comment]
Sorry for the trouble. I might try again to make the current version work later on. I'll let you know when I find anything interesting. I am suspecting my custom jekyll collections might cause the problem.

**Comment by christian-fei** [comment]
No worries, had the chance to refresh the code base a little bit! :+1:

**Comment by jtebert** [comment]
I seem to be having this same issue. I just updated from using the minified script to using this through Bower (v 1.1.1). No matter what text I put into the search box, it always returns all results. I created a [minimal test page to demonstrate the problem](http://juliaebert.com/jekyll-polymer/test-search/). I've done a bit of stepping through the source code/debugging but haven't found the source of the problem yet.

**Comment by christian-fei** [comment]
Hey @jtebert, thanks for the test page. just tried changing fuzzy to false and it works as expected.
Sorry for the trouble.
If you have some time I would love suggestions on how to improve the strategy throught a PR.
I see what I can do :)

Thanks

**Comment by jtebert** [comment]
I don't exactly understand what the fuzzy search is doing here. It seems like it just matches everything.

**Comment by christian-fei** [comment]
@jtebert there is a v1.1.2 on npm that could fix this problem

could someone please try it and report back?

https://www.npmjs.com/package/simple-jekyll-search

**Comment by christian-fei** [comment]
check out v1.1.3

**Comment by junedev** [comment]
@christian-fei @jtebert I tried the v1.1.4 (non-minified). The search seems fine with fuzzy: false but when I set it to true it is not working correctly. When I type "a" I get everything as a result. When I type "aaaaa" I still get everything. When I add other letters of the alphabet "aaaaasdfhkz" I get less and less results the more letters I add. I am matching with two custom collections instead of regular posts so I am not entirely sure this is representative for the usual use case.
